source/data-formats/modeling-bson-data.txt

```diff
@@ -11,140 +11,198 @@ Modeling BSON Data
+to the :ref:`sample BSON document <php-bson-sample>`:
 
-Consider the following class definition:
+- Adds a new ``restaurant_id`` field that has a value of ``12345``
+- Sets the value of the ``name`` field to ``"Mongo's Pizza Place"``
 
-.. code-block:: php
+.. literalinclude:: /includes/bson.php
+    :language: php
+    :dedent:
+    :start-after: start-modify-doc
+    :end-before: end-modify-doc
 
-   <?php
+Customize BSON Serialization
+----------------------------
 
-   class Person implements MongoDB\BSON\Persistable
-   {
-       private MongoDB\BSON\ObjectId $id;
-       private string $name;
-       private MongoDB\BSON\UTCDateTime $createdAt;
-
-       public function __construct(string $name)
-       {
-           $this->id = new MongoDB\BSON\ObjectId;
-           $this->name = $name;
-           $this->createdAt = new MongoDB\BSON\UTCDateTime;
-       }
-
-       function bsonSerialize()
-       {
-           return [
-               '_id' => $this->id,
-               'name' => $this->name,
-               'createdAt' => $this->createdAt,
-           ];
-       }
-
-       function bsonUnserialize(array $data)
-       {
-           $this->id = $data['_id'];
-           $this->name = $data['name'];
-           $this->createdAt = $data['createdAt'];
-       }
-   }
+The following sections describe how to configure your application's 
+serialization of BSON data:
 
-The following example constructs a ``Person`` object, inserts it into the
-database, and reads it back as an object of the same type:
+- :ref:`php-type-map`: Use the ``typeMap`` option to specify the default conversion
+  between {+language+} types and BSON types
+
+- :ref:`php-persistable-classes`: Use the ``MongoDB\BSON\Persistable`` interface to
+  enable serialization
+
+- :ref:`php-enums`: Use the ``bsonSerialize()`` and ``bsonUnserialize()`` methods to
+  specify serialization between backed enums and BSON values
+
+.. _php-type-map:
+
+Type Maps
+~~~~~~~~~
+
+You can set the ``typeMap`` option, which configures how BSON values are converted to
+{+language+} values, at the following levels:
+
+- ``MongoDB\Client``, which sets the *default for all operations* unless overridden
+- ``MongoDB\Database``
+- ``MongoDB\Collection``
+
+This list also indicates the increasing order of precedence of the option settings. For
+example, if you set a ``typeMap`` for a collection, it will override the type map
+set on the database.
+
+The {+library-short+} uses the following type map by default:
 
 .. code-block:: php
 
-   <?php
+   [
+       'array' => 'MongoDB\Model\BSONArray',
+       'document' => 'MongoDB\Model\BSONDocument',
+       'root' => 'MongoDB\Model\BSONDocument',
+   ]
 
-   $collection = (new MongoDB\Client)->test->persons;
+This type map performs the following conversions:
 
-   $result = $collection->insertOne(new Person('Bob'));
+- Arrays to ``MongoDB\Model\BSONArray`` objects
+- Top-level and embedded BSON documents to ``MongoDB\Model\BSONDocument`` objects
```

Comment:
S: clarify that these apply in both directions (serialization & deserialization)

Reply:
This feedback applies to wording in other parts of this section as well